### ISSUE TITLE
Does syslog require registration to be used via spdlog::get("syslog") ?

### ISSUE THREAD

**Issue by MaximAxelrod** [gabime/spdlog#1839]
Hi All, 
I need to use syslog in my project. 
I have seen the following example:
```
#include "spdlog/sinks/syslog_sink.h"
void syslog_example()
{
    std::string ident = "spdlog-example";
    auto syslog_logger = spdlog::syslog_logger_mt("syslog", ident, LOG_PID);
    syslog_logger->warn("This is warning that will end up in syslog.");
}
```

If I am doing what is proposed in this example, can I use syslog with spdlog::get("syslog") without calling to `spdlog::register_logger(syslog_logger);`? 

If I am trying to register syslog_logger, I am getting an exception that the syslog is already registered.

Am I doing something wrong? if yes, can you please help me, what is the proper way to archive the following behavior: 
`spdlog::get("syslog")->log(severity, sting_buf);` ?

**Comment by tt4g** [comment]
> If I am doing what is proposed in this example, can I use syslog with spdlog::get("syslog") without calling to spdlog::register_logger(syslog_logger);?

Yes, you must not call `spdlog::register_logger(syslog_logger)`, because the internal processing of `spdlog::syslog_logger_mt("syslog", ident, LOG_PID)` is equivalent to that of `spdlog::register_logger(syslog_logger)`.

After calling `spdlog::syslog_logger_mt("syslog", ident, LOG_PID);`, `auto syslog_logger = spdlog::get("syslog")` should be able to be called.

By the way, `spdlog::get("syslog")->log(severity, sting_buf);` is not correct.
The first argument should be `spdlog::level::level_enum`.

**Comment by MaximAxelrod** [comment]
Thank you so much!
